internal/dialect/dialectquery/dialectquery.go

```diff
@@ -26,3 +26,9 @@ type Querier interface {
 	// The query should return the version_id and is_applied columns.
 	ListMigrations(tableName string) string
 }
+
+// QuerierOptions is an interface to provide specific options to a dialect.
+// For ex: providing replication support to underlying migration table.
+type QuerierOptions interface {
```

Comment:
When I initially added a "store" and "querier" the idea was to have the core goose library only interact with the store, and have no idea about the underlying queries.

The store abstracts the underlying querier which holds the raw string queries for the different database technologies. 

Not sure if this made it easier/harder to evolve, but we should probably think about if there's a more elegant way to now thread options from the core library to those 2 internal layers.

Reply:
Yeah, I struggled around this as well. I didn't find an easier way to hookup options all the way to the implementation specific detail. I tried to minimize the leaky abstraction(s) but I think it will have to exposed in one way.

If you have any suggestions, I can try those out.